### ISSUE TITLE
last arc from .arcs() should be tangent to handle

### ISSUE THREAD

**Issue by deanmartinez** [Pomax/bezierjs#72]
I ran into a case when getting .arcs() that the last arc was not tangent to the handle. Given this case:

```
new Bezier([25.308000000000003,10.260000000000001,25.848000000000002,10.728000000000001,25.848000000000002,11.304000000000002]).arcs(0.0012143080752705958)
```

The control point and end point have the same X value, `25.848000000000002` so the handle is vertical. So I am expecting that the last arc's end angle should be zero. Instead it's about 4 degrees.

**Comment by Pomax** [comment]
This does depend on the error threshold, though, so one question is: does it change when we change the error threshold here?

**Comment by deanmartinez** [comment]
Yes, with a different threshold I can get different results.

**Comment by Pomax** [comment]
I'll be honest: even blown up to 100x those look like perfectly fine arc approximation at that error threshold: http://jsbin.com/sehemelusi/edit?js,output

**Comment by Pomax** [comment]
I've pushed a v2.4.4 which makes the previous jsbin render correctly: the arc now has start and end angles that line up with where they should be, even when the final angle leads to capping.

**Comment by deanmartinez** [comment]
Awesome, thank you! I've ran a few tests that failed previously, now are working great!

**Comment by deanmartinez** [comment]
Oh, I just ran into a couple of cases where I get an arc that is not even within the bounding box.
var pts = [
  2232.0000000000004,3348.0000000000004,
  2656.8,3348.0000000000004,
  3002.4000000000004,3132.0000000000004
];
var threshold = 0.8109375920992841;

**Comment by deanmartinez** [comment]
I was also able to repro in the API page:
![image](https://cloud.githubusercontent.com/assets/25230619/24737816/7df16e2e-1a46-11e7-8f52-c81cfd5f1993.png)

**Comment by deanmartinez** [comment]
And saw an error in the console: 
Uncaught TypeError: Cannot create property 's' on boolean 'false'
    at Object.getccenter (bezier.js:1)
    at l._iterate (bezier.js:1)
    at l.arcs (bezier.js:1)
    at draw (<anonymous>:6:20)
    at Object.handleInteraction.onupdate (<anonymous>:15:81)
    at HTMLCanvasElement.<anonymous> (interaction.js:55)

**Comment by Pomax** [comment]
good find, that error would come from https://github.com/Pomax/bezierjs/blob/gh-pages/lib/utils.js#L540 which would suggest that https://github.com/Pomax/bezierjs/blob/gh-pages/lib/utils.js#L519 is yielding a `false`... which implies a line/line intersection test is failing pretty hard over in https://github.com/Pomax/bezierjs/blob/gh-pages/lib/utils.js#L203-L209 (specifically, `d` is zero, which suggests parallel lines...)

It will be worth getting some coordinates figured out for reliable testing.

**Comment by deanmartinez** [comment]
So what does utils.angle give? Looks like it takes 3 points as parameters...

**Comment by deanmartinez** [comment]
Ok I found some coords that repro: 172, 158, 113, 89, 29, 82

![image](https://cloud.githubusercontent.com/assets/25230619/24826242/227268ae-1be6-11e7-8273-e04f542fe1cc.png)

**Comment by deanmartinez** [comment]
These repro the arc out of the bbox:
172, 158, 108, 124, 29, 82
168, 116, 106, 96, 57, 129
180, 113, 112, 123, 37, 134
161, 147, 114, 86, 44, 78

This repros the "parallel" bug. Probably because the curve is linear. I suppose that could be expected:
168, 68, 106, 73, 44, 78

**Comment by Pomax** [comment]
yeah there is no linearity detection at the moment, so I'm okay with those yielding bad arcs. The others are more interesting, I'll make some coffee and investigate what is going on. Thanks for the help in finding some good test cases!

**Comment by Pomax** [comment]
Running your test cases through the jsbin, above, I get:

`172, 158, 108, 124, 29, 82` appears to be linear but does not show any odd arcs:

![screenshot 214](https://cloud.githubusercontent.com/assets/177243/24831062/802c5f16-1c47-11e7-8e83-70cb2cbd8af0.png)

`168, 116, 106, 96, 57, 129` looks fine far as I can tell:

![screenshot 213](https://cloud.githubusercontent.com/assets/177243/24831056/7034643c-1c47-11e7-8fbf-bec36240abea.png)

`180, 113, 112, 123, 37, 134` appears to be linear but does not show any odd arcs:

![screenshot 215](https://cloud.githubusercontent.com/assets/177243/24831068/a13026de-1c47-11e7-9835-adfe99a9c9da.png)

`161, 147, 114, 86, 44, 78` looks fine far as I can tell:

![screenshot 216](https://cloud.githubusercontent.com/assets/177243/24831076/be954970-1c47-11e7-8e60-d28c2a0e7345.png)

`147, 74, 110, 80, 59, 108, 46, 169` looks super wrong:

![image](https://cloud.githubusercontent.com/assets/177243/24831079/cd6e0a40-1c47-11e7-9636-ae07b644cd2e.png)

`169, 140, 137, 107, 83, 89, 34, 94` also looks super wrong:

![image](https://cloud.githubusercontent.com/assets/177243/24831085/df274918-1c47-11e7-98ac-a363798fd420.png)

**Comment by Pomax** [comment]
Just in case this is caused by maths engines: which OS/Browser are you seeing things go wrong on, and do you have code that you can drop in a JSbin that shows off the error?

**Comment by deanmartinez** [comment]
I'm using your API page, in Chrome. I will try to make a test page with some of these values.

**Comment by deanmartinez** [comment]
Oh, since I was using the API page, that must mean that the default threshold was used, 0.5

**Comment by deanmartinez** [comment]
Ok, I can repro in the JsBin - just set threshold to null :)

**Comment by deanmartinez** [comment]
Also found if helpful to add this to the `arcs.forEach` drawing:
```
ctx.lineTo(a.x,a.y);
```
so you can visualize the very skinny arc, in those cases which appear to be linear.

**Comment by Pomax** [comment]
Ah! good point, using 0.5 seems to be a magical value for the threshold.
